src/fusedav.c

```diff
@@ -290,59 +372,75 @@ static void getdir_propfind_callback(__unused void *userdata, const char *path,
+                        funcname, status_code, path);
+                delete_path_on_propfind(path, &subgerr1);
+                if (subgerr1) {
+                    g_propagate_prefixed_error(gerr, subgerr1, "%s: ", funcname);
+                }
             }
+            // fileserver thinks file exists. If we got a 410, we could just keep things the way they were.
+            // But we'd have to check that what was local wasn't a negative entry.
+            // Seems wise to just create in all cases
             else {
-                // REVIEW: if the file exists, do we want to call stat_cache_value_set and update its ->updated value?
                 if (response_code >= 200 && response_code < 300) {
-                    // We're not deleting since it exists; jump out!
                     log_print(LOG_NOTICE, SECTION_FUSEDAV_PROP, 
-                            "%s: saw 410; executed HEAD; file exists: %s", 
-                            funcname, path);
+                            "%s: saw %lu; executed HEAD; file exists: %s", 
+                            funcname, status_code, path);
+                    create_path_on_propfind(path, st, &subgerr1);
+                    if (subgerr1) {
+                        g_propagate_prefixed_error(gerr, subgerr1, "%s: ", funcname);
+                    }
                 }
                 else {
-                    // On error, prefer retaining a file which should be deleted over deleting a file which should be retained
+                    // On error, retain local file
                     g_set_error(gerr, fusedav_quark(), EINVAL,
-                        "%s(%s): saw 410; HEAD returns unexpected response from curl %ld",
-                        funcname, path, response_code);
+                        "%s(%s): saw %lu; HEAD returns unexpected response from curl %ld",
+                        funcname, path, status_code, response_code);
                 }
-                return;
             }
         }
-
-        log_print(LOG_INFO, SECTION_FUSEDAV_PROP, "%s: Removing path: %s", 
-                funcname, path);
-        stat_cache_delete(config->cache, path, &subgerr1);
-        filecache_delete(config->cache, path, true, &subgerr2);
-        // If we need to combine 2 errors, use one of the error messages in the propagated prefix
-        if (subgerr1 && subgerr2) {
-            g_propagate_prefixed_error(gerr, subgerr1, "%s: %s :: ", funcname, subgerr2->message);
-        }
-        else if (subgerr1) {
-            g_propagate_prefixed_error(gerr, subgerr1, "%s: ", funcname);
-        }
-        else if (subgerr2) {
-            g_propagate_prefixed_error(gerr, subgerr2, "%s: ", funcname);
-        }
     }
+    // 5. local entry is older than what propfind is showing
+
+    /* if local entry exists but ctime is older than what propfind indicates, use
+     * what propfind gives us. On 410 delete, even if what exists is already
```

Comment:
Looks like a verb is missing here.

Reply:
I think just a comma, "On 410, delete, ..."